### PR TITLE
Add additional functionality in REST layer for checking permissions.

### PR DESCRIPTION
**Pull request by michaelarusso**

### CODE REVIEW

**Comment by snoopdave** [comment]
Looks good overall, but why do we need to disable read-repairs?

If we have to disable read repairs we should mark the tests that fail (see below) with the Ignore annotation and a comment that explains it.

Failed tests:
  ServiceRequestIT.testRepairOfSingleEntity:160 Create call should have worked
  ServiceRequestIT.testRepairOfMultipleEntitiesAndRemainingEntities:313 null

**Comment by michaelarusso** [comment]
Read repair is being temporarily removed until we can test/validate further.  Good catch on the tests.  Fixing that now.

File: stack/rest/src/main/java/org/apache/usergrid/rest/applications/ApplicationResource.java [hunk 1]
```diff
@@ -38,6 +38,7 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriInfo;
 
+import org.apache.usergrid.rest.security.annotations.CheckPermissionsForPath;
```
Comment (mdunker) [inline stack/rest/src/main/java/org/apache/usergrid/rest/applications/ApplicationResource.java]: this is not used in ApplicationResource.java

Comment (michaelarusso) [inline stack/rest/src/main/java/org/apache/usergrid/rest/applications/ApplicationResource.java]: fixed.

**Pull request merged**